### DISCUSSION TITLE
Q&A - How to configure launch menu to navigate to a different folder

### DISCUSSION THREAD

**Discussion by minhquana3** [wezterm/wezterm#4570] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

Hello. I'm new to Wezterm and terminal emulator. I need your help to solve my below problem. I have tried to read the docs several times, but I can't manage to make it work. 

I want to set up an item in the launch menu for navigating to a different folder so that I can quickly navigate to the location without typing the folder path. 

The error I keep getting is "Process "cd" in domain "local" didn't exit cleanly. Exited with code 1. This message is shown because exit_behavior="CloseOnCleanExit"

My lua config is
```lua
local wezterm = require 'wezterm'
local launch_menu = {}

local config = {}

if wezterm.config_builder then
	config = wezterm.config_builder()
end

config.default_prog = { "C:/Program Files/Git/bin/bash.exe" }
config.window_decorations = "RESIZE"

table.insert(launch_menu, {
	label = "Move to the Project folder",
	args = { "cd", },
	cwd = "/D/Projects/Go/Web",
})

config.keys = {
	{
		key = 'l',
		mods = 'ALT',
		action = wezterm.action.ShowLauncher
	}
}

config.launch_menu = launch_menu

return config
```

**Comment by wez** [comment, marked as answer]
The launch menu is used to start a new program.
If you want to start a new bash shell in a particular directory, you probably want:

```lua
table.insert(launch_menu, {
	label = "New shell in Project folder",
	args = { "C:/Program Files/Git/bin/bash.exe" },
	cwd = "/D/Projects/Go/Web",
})
```

If you want to make the current shell change directory, you can't do that from the launch menu.  You can instead define a key assignment that sends the `cd /D/Projects/Go/Web` command to the active pane:

```
config.keys = {
  { key = 'm', mods = 'ALT', action = wezterm.action.SendString 'cd /D/Projects/Go/Web\n' },
}
```

key bindings like this also show up in the [command palette](https://wezfurlong.org/wezterm/config/lua/keyassignment/ActivateCommandPalette.html).

If you'd prefer not to assign keys to this, you can instead use the [augment-command-palette](https://wezfurlong.org/wezterm/config/lua/window-events/augment-command-palette.html) event to add entries without key assignments.

**Reply by minhquana3** [reply]
The key assignment worked for me. Thank you for answering it.
